src/codeu/chat/client/Controller.java

```diff
@@ -89,6 +90,31 @@ public User newUser(String name) {
     return response;
   }
 
+
+  public User restoreUser(User user) {
+
+    User response = null;
+
+    try (final Connection connection = source.connect()) {
+
+      Serializers.INTEGER.write(connection.out(), NetworkCode.NEW_USER_REQUEST);
```

Comment:
The javadoc on this method may clarify, but is it intentional to use the NEW_USER codes for this restoreUser method?